pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -461,7 +461,8 @@ protected CompletableFuture<Void> doAcknowledge(MessageId messageId, AckType ack
             Consumer individualConsumer = consumers.get(topicMessageId.getTopicPartitionName());
             if (individualConsumer != null) {
                 MessageId innerId = topicMessageId.getInnerMessageId();
-                return individualConsumer.acknowledgeCumulativeAsync(innerId);
+                return individualConsumer.acknowledgeCumulativeAsync(innerId)
+                        .thenAccept(__ -> unAckedMessageTracker.remove(topicMessageId));
```

Comment:
cumulativeAck, not every message will invoke, so use remove can't remove all. if use removeMessagsTill(), all partitioned topics will be removed, so its better to find a better way to handle this case